python/cog/predictor.py

```diff
@@ -92,9 +95,14 @@ class Input(BaseModel):
     for name, parameter in signature.parameters.items():
         InputType = parameter.annotation
 
-        if not InputType:
-            # TODO: perhaps should throw error if there are arguments not annotated?
-            continue
+        if InputType == inspect.Signature.empty:
+            raise TypeError(
+                f"No input type provided for parameter `{name}`. Supported input types are: {get_readable_type_names(ALLOWED_INPUT_TYPES)}."
+            )
+        elif InputType not in ALLOWED_INPUT_TYPES:
+            raise TypeError(
+                f"Unsupported input type {get_type_name(annotation)} for parameter `{name}`. Supported input types are: {get_readable_type_names(ALLOWED_INPUT_TYPES)}."
```

Comment:
Where does `annotation` come from?

Reply:
Argh, thank you. That would've gone wrong in a rebase – `InputType` used to be called `annotation`.

The fact this passes CI rather suggests there's no test coverage here, so I might add one.

Reply:
Added a test (reused the old one I'd previously removed, in fact) and fixed the error.

Reply:
Nice!